projects/node-http-exercise/node-http-exercise.md

```diff
@@ -0,0 +1,255 @@
+
+3. Then, still using any variable name you want, create default content-type as `application/json`.
+
+4. Set a default response object as such:
+
+```javascript
+let response = {
+  message: 'Not Found',
+};
+```
+
+5. Set an empty header object:
+
+```javascript
+let header = {};
+```
+
+6. Refactor the writing header function to use the placeholders:
+
+```javascript
+res.writeHead(status, { 'Content-Type': contentType });
+```
+
+7. Use the `JSON.stringify()` function to parse the response object and send it back in your `.end()`:
+
+```javascript
+res.end(JSON.stringify(response));
+```
+
+Now if you test your API with the browser by going to the same url as above, you should get this as a response:
+
+```text
+{"message":"Not Found"}
+```
+
+8. Outside of the server, create your routes object, populating it with a test route that you will remove later on. I usually place it before the `server.listen()` function:
+
+```javascript
+const routes = {
+  '/test': {
+    status: 200,
+    contentType: 'text/html',
+    message: 'It Works!',
+  },
+};
+```
+
+9. Create the helper function to construct the response object depending on the endpoint.
+
+```javascript
+for (const currentRoute in routes) {
+  if (pathname === currentRoute) {
+    // pull out this route's response
+    let routeResponse = routes[currentRoute];
+
+    // set status to response's or to default
+    status = routeResponse.status || status;
+
+    // set contentType to response's or to default
+    contentType = routeResponse.contentType || contentType;
+
+    // if this route has a message, build response with it
+    if (routeResponse.message) {
+      response.message = routeResponse.message;
+    }
+
+    // if this route has a custom response, use it
+    if (routeResponse.response) {
+      response = routeResponse.response;
+    }
+
+    // if this route has custom headers, populate them for the response
+    if (routeResponse.header) {
+      header = { ...routeResponse.header };
+    }
+  }
+}
+```
+
+When testing this by going to `http://localhost:5000/test`, you should get:
+
+```text
+{"message":"It Works!"}
+```
+
+### Step 3 - Create Your Routes
+
+This is the core step of the assignment. Since you have a working structre, the endpoints creation should be easier.
+
+Now, we will build the following routes requested in your API. If you get stuck and need some help in creating the routes, you can look in the [hints](#hints) section for help.
+
+1. Create the `/spin` route, wich should make a GET request to the server and this one should respond with nothing, but keep on being open for connections.
+
+2. Create the `/hangup` route, which should close the connection with no response. This one can be created in multiple ways. Think about maybe making a check for this route before the helper function. Remember that you are looking for the `pathname` for this check. See [hints](#hints) for more information about this.
```

Comment:
When I wrote the assignment reqs I forgot that by default, there will be a 200 response code so that might warrant a hint that it must be unset, e.g. `res.statusCode = null` or we can just remove this case. What do you think @alodahl ?

Reply:
I added `status = 200`. Let me know if you want to remove entirely.

fixed in [799330c](https://github.com/Techtonica/curriculum/pull/1406/commits/799330cf2b82a0e0d74d3cd2e258f97adc3e70db)